test/e2e/status_subresource_test.go

```diff
@@ -52,3 +57,73 @@ func testFinalizerWhenStatusForConfigResourcesEnabled(t *testing.T) {
 	err = framework.DeletePrometheusAndWaitUntilGone(ctx, ns, name)
 	require.NoError(t, err, "failed to delete Prometheus with status-cleanup finalizer")
 }
+
+// testServiceMonitorStatusSubresource validates ServiceMonitor status updates upon Prometheus selection.
+func testServiceMonitorStatusSubresource(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(ctx, t, testCtx)
+	framework.SetupPrometheusRBAC(ctx, t, testCtx, ns)
+	_, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(
+		ctx, testFramework.PrometheusOperatorOpts{
+			Namespace:           ns,
+			AllowedNamespaces:   []string{ns},
+			EnabledFeatureGates: []operator.FeatureGateName{operator.StatusForConfigurationResourcesFeature},
+		},
+	)
+	require.NoError(t, err)
+	name := "servicemonitor-status-subresource-test"
+
+	p := framework.MakeBasicPrometheus(ns, name, name, 1)
+
+	_, err = framework.CreatePrometheusAndWaitUntilReady(ctx, ns, p)
+	require.NoError(t, err, "failed to create Prometheus")
```

Comment:
(nit) IMHO, additional test messages are useful when they provide contextual information which helps troubleshooting. In this case when the assertion isn't met, the test output will display which line failed which is enough to understand the context.
```suggestion
	require.NoError(t, err)
```